validator/rules/executor.go

```diff
@@ -46,35 +46,42 @@ func executeHelper(cmd *cobra.Command, info *validator.StatusLog, ruleConfig *Ru
 
 // executeRecursive recursively traverse over all the subcommands
 // and validate using executeRulesChildren function
-func executeRecursive(cmd *cobra.Command, info *validator.StatusLog, ruleConfig *RuleConfig) []validator.ValidationError {
+func executeRecursive(cmd *cobra.Command, info *validator.StatusLog, ruleConfig *RuleConfig, userValidatorConfig *ValidatorConfig) []validator.ValidationError {
 	for _, child := range cmd.Commands() {
 		// base case
 		if !child.IsAvailableCommand() || child.IsAdditionalHelpTopicCommand() {
 			continue
 		}
 		// recursive call
-		info.Errors = executeRecursive(child, info, ruleConfig)
+		info.Errors = executeRecursive(child, info, ruleConfig, userValidatorConfig)
 	}
-	info.Errors = executeRulesChildren(cmd, info, ruleConfig)
+	info.Errors = executeRulesChildren(cmd, info, ruleConfig, userValidatorConfig)
 
 	return info.Errors
 }
 
 // executeRulesChildren execute rules on children of cmd
-func executeRulesChildren(cmd *cobra.Command, info *validator.StatusLog, ruleConfig *RuleConfig) []validator.ValidationError {
+func executeRulesChildren(cmd *cobra.Command, info *validator.StatusLog, ruleConfig *RuleConfig, userValidatorConfig *ValidatorConfig) []validator.ValidationError {
 	children := cmd.Commands()
 	for _, child := range children {
 
 		if !child.IsAvailableCommand() || child.IsAdditionalHelpTopicCommand() {
 			continue
 		}
-		validate(child, info, ruleConfig)
+		validate(child, info, ruleConfig, userValidatorConfig)
 	}
 	return info.Errors
 }
 
 // validate returns validation errors by executing the rules
-func validate(cmd *cobra.Command, info *validator.StatusLog, ruleConfig *RuleConfig) {
+func validate(cmd *cobra.Command, info *validator.StatusLog, ruleConfig *RuleConfig, userValidatorConfig *ValidatorConfig) {
+
+	// if command needs to be ignored
+	if val, ok := userValidatorConfig.ValidatorOptions.IgnoreCommands[cmd.Use]; ok {
```

Comment:
How this will work with nested commands? Do we use `.` to map the nested commands.

Example:
```
kafka.create 
registry.create
```

Reply:
it captures `Use` of the command, so 
```
kafka.create 
registry.create
```
all `create` subcommands will be ignored. Let's fix this!

Reply:
However would that be hard. Can we simply create method that returns true or false for the current command and rule rule?

Reply:
for nested commands, best way is to use CommandPath() provided by cobra. 
user will specify : `{"kafka create": true, "echo": true}`

This config is coming to my mind atm -
```go
type ValidatorOptions struct {
	Verbose      bool            
	SkipChildren map[string]bool 
	SkipCommands map[string]bool 
}
```
RuleOptions to be present in each rule, for controlling rule specific properties/options
```go
type RuleOptions struct {
	Verbose      bool
	SkipCommands map[string]bool
}
```
eg - 
```go
type Length struct {
	RuleOptions validator.RuleOptions
	Limits      map[string]Limit `json:"Limits"`
}

Reply:
type Length struct {
	RuleOptions validator.RuleOptions
	
Could that be part of the interface so we can still have generic processing?

Reply:
> RuleOptions validator.RuleOptions
Could that be part of the interface so we can still have generic processing and enforce options for each rule

Rules interface?

> SkipChildren map[string]bool 
Not needed/confusing. If you specify command path that is grouping other commands all grouped ones will be disabled.

can you please provide an example for understanding

> map[string]bool 
> Not sure if that would give us enough to control command (path to command), rule, level - Error/Ignore (could be warn in the future)

do we need all this details, when we ignore some command?

Reply:
> do we need all this details, when we ignore some command?

Agreed. No need